RFPStrategy/RFPMilestone/spec.ts

```diff
@@ -0,0 +1,78 @@
+import { Address, BeforeAll, Event, LiveObject, OnEvent, Property, saveAll,Spec } from '@spec.dev/core'
+
+import { getStatusFromInt } from '../../shared/status'
+
+/**
+ * RFP Milestone details
+ */
+@Spec({
+    uniqueBy: ['strategyId', 'id', 'chainId']
+})
+class RFPMilestone extends LiveObject {
+
+    @Property()
+    milestoneId: number
+
+    @Property()
+    strategyId: Address
+
+    @Property()
+    metadataProtocol: number
+
+    @Property()
+    metadataPointer: stringg
+
+    @Property()
+    status: string
+
+    // ====================
+    // =  Event Handlers  =
+    // ====================
+
+    @BeforeAll()
+    setCommonProperties(event: Event) {
+        this.strategyId = event.origin.contractAddress;
+        this.milestoneId = event.data.milestoneId;
+    }
+
+    @OnEvent('allov2.RFPSimple.MilestonesSet')
+    async onMilestonesSet(event: Event) {
+        // TODO-SPEC: figure out fetch all milestones from strategyId and mark them as deleted state in status
+        // when invoked the second time
+        // this is for soft delete ^
+
+        const strategyContract = this.bind(this.address, 'allov2.RFPSimple')
```

Comment:
You can actually just use `this.contract` instead of creating a new variable for this. `this.contract` is an already established binding to the contract that emitted the current event you're handling.